clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -3331,6 +3343,65 @@ checkBuiltinTemplateIdType(Sema &SemaRef, BuiltinTemplateDecl *BTD,
     QualType HasNoTypeMember = Converted[2].getAsType();
     return HasNoTypeMember;
   }
+  case BTK__builtin_dedup_pack: {
+    assert(Converted.size() == 1 && "__builtin_dedup_pack should be given "
+                                    "a parameter pack");
+    TemplateArgument Ts = Converted[0];
+    // Delay the computation until we can compute the final result. We choose
+    // not to remove the duplicates upfront before substitution to keep the code
+    // simple.
+    if (Ts.isDependent())
+      return QualType();
+    assert(Ts.getKind() == clang::TemplateArgument::Pack);
+    llvm::SmallVector<TemplateArgument> OutArgs;
+    llvm::SmallDenseSet<QualType> Seen;
+    // Synthesize a new template argument list, removing duplicates.
+    for (auto T : Ts.getPackAsArray()) {
```

Comment:
For this array, should we just do an insert into a MapVector instead?

Reply:
I couldn't make it work. The defaults gives me `SmallVector<pair<QualType, TemplateArgument>>`, and if I specify `MapVector<QualType, TemplateArgument, DenseMap<QualType, unsigned>, SmallVector<TemplateArgument>>`, `try_emplace` fails because it still tries to construct a pair inside the vector.

Am I holding it wrong?

Reply:
I suspect it is a 'holding it wrong'?  Though I probably MEANT to suggest `SetVector`?  Does that make it easier?